mantidproject/mantid
# Output optimal fit function after completion of fitting

**To test:**

Download and unzip [test_files.zip](https://github.com/mantidproject/mantid/files/1246491/test_files.zip) under `/tmp` and run the script:

```
from __future__ import print_function
import numpy as np

resolution_file = '/tmp/BASIS_59304_divided.dat'
data_file = '/tmp/BASIS_59293_divided.dat'
resolution = LoadDaveGrp(Filename=resolution_file, OutputWorkspace='resolution', IsMicroEV=True)
data = LoadDaveGrp(Filename=data_file, OutputWorkspace='data', IsMicroEV=True)

function = '''(composite=Convolution,FixResolution=false,NumDeriv=true;
name=TabulatedFunction,Workspace=resolution,WorkspaceIndex=0,Scaling=0.5,Shift=0,XScaling=1,ties=(XScaling=1);
(name=DeltaFunction,Height=0.5,Centre=0,constraints=(0<Height<1),ties=(Centre=0);
name=Lorentzian,Amplitude=0.5,PeakCentre=0,FWHM=0.02,constraints=(0.001<FWHM),ties=(PeakCentre=0);ties=(f1.Amplitude=1-f0.Height)));
name=LinearBackground,A0=0,A1=0
'''

results = Fit(Function=function, InputWorkspace=data, WorkspaceIndex=0, CreateOutput=True)
print(results.Function)
```
The printed function should be
```
(composite=Convolution,FixResolution=false,NumDeriv=true;name=TabulatedFunction,Workspace=resolution,WorkspaceIndex=0,X=(),Y=(),Scaling=1.09281,Shift=1.47731e-10,XScaling=1,ties=(XScaling=1);(name=DeltaFunction,Height=0.87475,Centre=0,constraints=(0<Height<1),ties=(Centre=0);name=Lorentzian,Amplitude=0.12525,PeakCentre=0,FWHM=0.00874458,constraints=(0.001<FWHM),ties=(PeakCentre=0);ties=(f1.Amplitude=1-f0.Height)));name=LinearBackground,A0=3.41084e-05,A1=6.46268e-05
```

Fixes #20326
[Release Notes](https://github.com/mantidproject/mantid/blob/1d049fc61cc0b58e75dfa10546811733107a01ca/docs/source/release/v3.11.0/framework.rst)

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
